[Docs-2.0] `TechnicalOverview.md` into `FutureOfWasabiWallet.md` (#1029)
zkSNACKs/WasabiDoc: docs/building-wasabi/FutureOfWasabiWallet.md

---
{
  "title": "Future Of Wasabi Wallet",
  "description": "This document aims to describe some ideas for the future of Wasabi Wallet and to explain the reasoning behind going towards these objectives. Plans might change without notice and this only represents our thinking at 2021 December. This is the Wasabi documentation, an archive of knowledge about the open-source, non-custodial and privacy-focused Bitcoin wallet for desktop.",
  "sidebarDepth": "1"
}
---

# Ideas And Plans For The Future Of Wasabi Wallet

---

> Wasabi Wallet is a privacy focused Bitcoin wallet that is based on the [WabiSabi Framework](https://eprint.iacr.org/2021/206).

Reviewer: _non-custodial_ might be good to add
```suggestion
Wasabi Wallet is a privacy focused, non-custodial Bitcoin wallet that is based on the [WabiSabi Framework](https://eprint.iacr.org/2021/206).
```

> While privacy can be achieved today with it, the mainchain costs, inconvenience and scalability will inevitably push us to look in to Lightning Network and other 2nd layer solutions.

Reviewer: > it

To me, it is not clear what you mean with it. It can mean multiple things

Author: I think it's clear it means WW even though it can mean both but tautology should be considered as bigger sin.

Author: Changed to even tho I think it was ok as it was.

This document aims to describe some ideas for the future of Wasabi Wallet and to explain the reasoning behind going towards these objectives. Plans might change without notice and this only represents our thinking at 2021 December.
Wasabi does not support and does not plan to support other currencies in the future.

> Wasabi is software. Therefore it must not neglect general software quality issues.

Reviewer: Otherwise Wasabi can be the hot spicy thingy aswell ;)
```suggestion
Wasabi Wallet is software. Therefore it must not neglect general software quality issues.
```

In Wasabi, this is most apparent during CoinJoins.
The more users that participate in a round, the higher the anonymity set the CoinJoin achieves, and the more frequent the rounds will be.

### CoinJoin

Even though Wasabi develoers have done a lot of work to optimize WabiSabi CoinJoins privacy and efficiency, the work is still in the beginning. 
For now, WabiSabi is only enabled for testnet clients, which means there is very little incentive for malicious attackers to reveal if they see a problem in the protocol.
Whenever mainchain CoinJoin release candidate comes out, we will likely see a lot of new testers with a bunch of issues and improvement ideas.

#### Send in CoinJoins

Even when a user has private coins, being able to send payments in CoinJoin is a great chain efficiency and privacy improvement for the sender. Change amount and maybe in some cases the payment amount should be grinded in to next lower standard denomination, making it very difficult to the recipient to figure out anything about the sender. 
GroupSend: [https://github.com/zkSNACKs/Meta/issues/29](https://github.com/zkSNACKs/Meta/issues/29)

#### Round Interoperability

> Because block size has limits and CoinJoin transaction has weight, we can't just add thousands of participants to a CoinJoin round. 

Reviewer: ```suggestion
Because block size has limits and a CoinJoin transaction has weight, we can't just add thousands of participants to a CoinJoin round. 
```

> Because block size has limits and CoinJoin transaction has weight, we can't just add thousands of participants to a CoinJoin round. 

Reviewer: > we

Might be better to not use _we_? Or explain who you mean with _we_

> Because block size has limits and CoinJoin transaction has weight, we can't just add thousands of participants to a CoinJoin round. 

Reviewer: ```suggestion
Because the block size has a limit and a CoinJoin transaction has weight, it is not practical to just add thousands of participants to a CoinJoin round. 
```

Author: I agree with not using "we". I'll rephrase the whole sentence.
